src/algebra/mod.rs

```diff
@@ -114,13 +123,13 @@ where
 /// We trust the implementation to have an associative addition.
 /// (this cannot be checked statically).
 pub trait MonoidValue:
-    Clone + Eq + 'static + HasZero + Add + AddByRef + AddAssign + AddAssignByRef
+    Clone + Eq + 'static + HasZero + Add<Output = Self> + AddByRef + AddAssign + AddAssignByRef
```

Comment:
Isn't this default?

Reply:
It is, but sometimes compiler becomes unsure.